datanode/networkhistory/snapshot/service_load_snapshot.go

```diff
@@ -497,10 +497,12 @@ func copyDataIntoDatabase(ctx context.Context, vegaDbConn sqlstore.Connection, c
 		return 0, fmt.Errorf("failed to disable triggers, setting session replication role to replica failed: %w", err)
 	}
 	defer func() {
-		_, err = vegaDbConn.Exec(ctx, "SET session_replication_role = DEFAULT;")
-		if err != nil {
-			err = fmt.Errorf("failed to re-enable triggers, setting session replication role to default failed: %w", err)
-			rowsCopied = 0
+		_, triggersErr := vegaDbConn.Exec(ctx, "SET session_replication_role = DEFAULT;")
+		if err == nil {
+			if triggersErr != nil {
```

Comment:
Wouldn't it be easier to have an && here?

if err == nil && triggersErr != nil